eng/tools/dependency-testing/index.js

```diff
@@ -49,34 +49,45 @@ function outputTestPath(projectFolderPath, sourceDir, testFolder) {
   console.log(`Emitted variable "PackageTestPath" with content: ${testPath}`)
 }
 
+async function updateScripts(testPackageJson, packageJsonContents){
+  if(packageJsonContents.scripts["integration-test:node"]){
+    var replaceWithTimeout = packageJsonContents.scripts["integration-test:node"].match(/--timeout [0-9]+/);
+    if(replaceWithTimeout !== null){
+      testPackageJson.scripts["integration-test:node"] = testPackageJson.scripts["integration-test:node"].replace(/--timeout [0-9]+/g, replaceWithTimeout);
+    }
+  }
+}
+
 async function insertPackageJson(repoRoot, packageJsonContents, targetPackagePath, targetPackageName, versionType, testFolder) {
   const testPath = path.join(targetPackagePath, testFolder);
   var templateJson = await packageUtils.readFileJson("./templates/package.json");
   var testPackageJson = templateJson;
-  testPackageJson.name = packageJsonContents.name.replace("@azure/", "azure-") + "-test";
+  if(packageJsonContents.name.startsWith("@azure/")){
+    testPackageJson.name = packageJsonContents.name.replace("@azure/", "azure-") + "-test";
+  }
+  else if(packageJsonContents.name.startsWith("@azure-rest/")){
+    testPackageJson.name = packageJsonContents.name.replace("@azure-rest/", "azure-rest-") + "-test";
+  }
+  await updateScripts(testPackageJson, packageJsonContents);
   testPackageJson.devDependencies = {};
   depList = {};
-  var projectFolder = path.basename(targetPackagePath);
-  var projectDir = path.basename(path.dirname(targetPackagePath));
   var allowedVersionList = {};
   depList[targetPackageName] = packageJsonContents.version;//works
   allowedVersionList[targetPackageName] = depList[targetPackageName];
   for (const package of Object.keys(packageJsonContents.dependencies)) {
     depList[package] = packageJsonContents.dependencies[package];
-    if (package.startsWith("@azure/")) {
       depList[package] = await findAppropriateVersion(package, packageJsonContents.dependencies[package], repoRoot, versionType);
       if (packageJsonContents.dependencies[package] !== depList[package]) {
         console.log(package);
         allowedVersionList[package] = depList[package];
         console.log(allowedVersionList[package]);
       }
-    }
   }
   testPackageJson.dependencies = depList;
 
   for (const package of Object.keys(packageJsonContents.devDependencies)) {
     testPackageJson.devDependencies[package] = packageJsonContents.devDependencies[package];
-    if (package.startsWith("@azure/")) {
+    if (package.startsWith("@azure/") || package.startsWith("@azure-rest/")) {
```

Comment:
For that I will have to read the package.json contents of the `package` - might increase the parsing time. I think it should be ok. Since I am anyways filtering out the utility packages - so not running min-max on the dev-tool and other utility packages.

Reply:
Alright, as long as we've got a list then it doesn't matter very much.